src/utils.jl

```diff
@@ -458,16 +458,46 @@ function flatten1(ex)
   for x in ex.args
     isexpr(x, :block) ? append!(ex′.args, x.args) : push!(ex′.args, x)
   end
-  # Don't use `unblock` to preserve line nos
+  # Don't use `unblock` to preserve line nos.
   return length(ex′.args) == 1 ? ex′.args[1] : ex′
 end
 
+false_or_bflatten(x) = x==false ? false : bflatten(x)
+
+function bflatten(x)
+    fx = flatten(x)
+    return isexpr(fx, :block) ? fx : Expr(:block,fx)
+end
```

Comment:
```suggestion
# Helper - only flatten the arguments of x
flatten_args(x::Expr) = Expr(x.head, map(flatten, x.args)...)
flatten_args(x) = x
```
I think you could replace all `bflatten` or `false_or_bflatten` calls with this? Like I mentioned in the other threads, I don't want to special-case `false`, I think that's mostly a Julia mistake. If we simply preserve `block`s inside of trys, it should be fine.

Reply:
The arguments of try have to be blocks, even if they contain only one argument. This would break that convention (and potentially break pretty printers and other macro code that relies on the convention). I can not seem to be able to force julia to parse a try expression into something that does not have the arguments either as blocks or as false. I think this is done on purpose, to disambiguate `finally false end` from complete absence of a finally.

```
julia> :(try; catch; else; finally; end) |> striplines |> dump
Expr
  head: Symbol try
  args: Array{Any}((5,))
    1: Expr
      head: Symbol block
      args: Array{Any}((0,))
    2: Bool false
    3: Expr
      head: Symbol block
      args: Array{Any}((0,))
    4: Expr
      head: Symbol block
      args: Array{Any}((0,))
    5: Expr
      head: Symbol block
      args: Array{Any}((0,))

julia> :(try; g() catch; h() else; f() finally; m() end) |> striplines |> dump
Expr
  head: Symbol try
  args: Array{Any}((5,))
    1: Expr
      head: Symbol block
      args: Array{Any}((1,))
        1: Expr
          head: Symbol call
          args: Array{Any}((1,))
            1: Symbol g
    2: Bool false
    3: Expr
      head: Symbol block
      args: Array{Any}((1,))
        1: Expr
          head: Symbol call
          args: Array{Any}((1,))
            1: Symbol h
    4: Expr
      head: Symbol block
      args: Array{Any}((1,))
        1: Expr
          head: Symbol call
          args: Array{Any}((1,))
            1: Symbol m
    5: Expr
      head: Symbol block
      args: Array{Any}((1,))
        1: Expr
          head: Symbol call
          args: Array{Any}((1,))
            1: Symbol f
```

Reply:
> The arguments of try have to be blocks, even if they contain only one argument. This would break that convention (and potentially break pretty printers and other macro code that relies on the convention). I can not seem to be able to force julia to parse a try expression into something that does not have the arguments either as blocks or as false. I think this is done on purpose, to disambiguate finally false end from complete absence of a finally.

Doesn't my suggested code preserve blocks? `Expr(x.head, map(flatten, x.args)...)`? Maybe I'm misunderstanding.